modules/stochastic_tools/src/surrogates/GaussianProcess.C

```diff
@@ -0,0 +1,103 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "GaussianProcess.h"
+#include "Sampler.h"
+
+#include "CovarianceFunctionBase.h"
+
+registerMooseObject("StochasticToolsApp", GaussianProcess);
+
+InputParameters
+GaussianProcess::validParams()
+{
+  InputParameters params = SurrogateModel::validParams();
+  params += CovarianceInterface::validParams();
+  params.addClassDescription("Computes and evaluates Gaussian Process surrogate model.");
+  return params;
+}
+
+GaussianProcess::GaussianProcess(const InputParameters & parameters)
+  : SurrogateModel(parameters),
+    CovarianceInterface(parameters),
+    _training_params(getModelData<RealEigenMatrix>("_training_params")),
+    _param_standardizer(getModelData<StochasticTools::Standardizer>("_param_standardizer")),
+    _data_standardizer(getModelData<StochasticTools::Standardizer>("_data_standardizer")),
+    _K(getModelData<RealEigenMatrix>("_K")),
+    _K_results_solve(getModelData<RealEigenMatrix>("_K_results_solve")),
+    _feproblem(*parameters.get<FEProblemBase *>("_fe_problem_base")),
```

Comment:
Are you using this fe_problem for anything?

Reply:
Note, that was moved to CovarianceInterface. Just forgot to remove. Good catch.